app/plugins/adp-backend/src/service/deliveryProgrammeRouter.ts

```diff
@@ -119,51 +135,69 @@ export async function createProgrammeRouter(
 
       const data: DeliveryProgramme[] = await deliveryProgrammesStore.getAll();
 
-      const isDuplicate: boolean = await checkForDuplicateTitle(
+      
+      const isDuplicateTitle: boolean = await checkForDuplicateTitle(
         data,
         req.body.title,
       );
-      if (isDuplicate) {
+
+      
+      const isDuplicateCode: boolean = await checkForDuplicateProgrammeCode(
+        data,
+        req.body.delivery_programme_code,
+      );
+
+      if (isDuplicateTitle) {
         res
           .status(406)
           .json({ error: 'Delivery Programme title already exists' });
-      } else {
-        const author = await getCurrentUsername(identity, req);
-        const deliveryProgramme = await deliveryProgrammesStore.add(
-          req.body,
-          author,
-        );
-        const programmeManagers = req.body.programme_managers;
-        if (programmeManagers !== undefined) {
-          const catalogEntities = await catalog.getEntities({
-            filter: {
-              kind: 'User',
-            },
-            fields: [
-              'metadata.name',
-              'metadata.annotations.graph.microsoft.com/user-id',
-              'metadata.annotations.microsoft.com/email',
-              'spec.profile.displayName',
-            ],
-          });
+        return; 
+      }
 
-          const catalogEntity: Entity[] = catalogEntities.items;
+      if (isDuplicateCode) {
+        res
+          .status(406)
+          .json({ error: 'Delivery Programme code already exists' });
+        return; 
+      }
+      const author = await getCurrentUsername(identity, req);
+      const deliveryProgramme = await deliveryProgrammesStore.add(
+        req.body,
```

Comment:
Maybe not a thing for this PR, but do we have any validation around the structure of the body here? Its typed as any if I remember correctly